examples/run_CPMG.py

```diff
@@ -10,116 +10,131 @@
+        ("delay_TTL", config_dict["deblank_us"]),
+        ("pulse_TTL", prog_p90_us, "ph_cyc", ph1_cyc),
+        ("delay", 
+            config_dict["tau_us"]
+            - 2 * prog_p90_us / pi
+            -marker_us
+            -config_dict['deblank_us']),
+        ("marker", "echo_label", config_dict["nEchoes"]),
+        ("delay_TTL", config_dict["deblank_us"]),
+        ("pulse_TTL", prog_p180_us, "ph_cyc", ph2_cyc),
+        ("delay", config_dict["deadtime_us"]),
+        ("acquire", config_dict["echo_acq_ms"]),
+        ("delay", 
+            config_dict['deadtime_us']
+            -2* marker_us
+            - config_dict['deblank_us']
+            ),
+        ("jumpto", "echo_label"),
+        ("delay", config_dict["repetition_us"]),
+    ],
+    nScans=config_dict["nScans"],
+    indirect_idx=0,
+    indirect_len=config_dict["nEchoes"],
+    adcOffset=config_dict["adc_offset"],
+    carrierFreq_MHz=config_dict["carrierFreq_MHz"],
+    nPoints=nPoints,
+    nEchoes = config_dict['nEchoes'],
+    acq_time_ms=config_dict["echo_acq_ms"],
+    SW_kHz=config_dict["SW_kHz"],
+    ret_data=None,
 )
-# }}}
-# {{{ chunk and save data
-if phase_cycling:
-    data.chunk("t", ["ph1", "echo", "t2"], [len(ph1_cyc), config_dict["nEchoes"], -1])
-    data.setaxis("ph1", ph1_cyc / 4)
-    data.setaxis("echo", r_[0 : config_dict["nEchoes"]])
-    if config_dict["nScans"] > 1:
-        data.setaxis("nScans", r_[0 : config_dict["nScans"]])
-    data.squeeze()
-    data.set_units("t2", "s")
-    fl.next("Raw - time")
-    fl.image(
-        data.C.mean("nScans"))
-    data.reorder("t2", first=False)
-    for_plot = data.C
-    for_plot.ft('t2',shift=True)
-    for_plot.ft(['ph1'], unitary = True)
-    fl.next('FTed data')
-    fl.image(for_plot.C.mean("nScans")
-    )
-else:
-    if config_dict["nScans"] > 1:
-        data.setaxis("nScans", r_[0 : config_dict["nScans"]])
-    data.rename("t","t2")
-    fl.next("Raw - time")
-    fl.image(
-        data.C.mean("nScans"))
-    data.reorder("t2", first=False)
-    for_plot = echo_data.C
-    for_plot.ft('t2',shift=True)
-    fl.next('FTed data')
-    fl.image(for_plot)
-data.name(config_dict["type"] + "_" + config_dict["cpmg_counter"])
-data.set_prop("postproc_type", "spincore_CPMGv2")
+## {{{ chunk and save data
+data.set_prop("postproc_type", "proc_CPMG_v2")
 data.set_prop("acq_params", config_dict.asdict())
-target_directory = getDATADIR(exp_type="ODNP_NMR_comp/CPMG")
+data.name(config_dict["type"] + "_" + str(config_dict["cpmg_counter"]))
+data.chunk(
+    "t",
+    ["ph_overall", "ph_diff", "nEcho", "t2"],
+    [len(ph_overall), len(ph_diff), int(config_dict["nEchoes"]), -1],
+)
+data.labels(
+    {
+        "nEcho": r_[0 : int(config_dict["nEchoes"])],
+        "ph_overall": r_[0 : len(ph_overall)],
+        "ph_diff": r_[0 : len(ph_diff)],
+    }
+)
+data.setaxis("ph_overall", ph_overall / 4)
+data.setaxis("ph_diff", ph_diff / 4)
+# }}}
+target_directory = getDATADIR(exp_type="ODNP_NMR_comp/Echoes")
```

Comment:
it seems like you frequently run into file saving errors.  Because of those, I would move this up before any instrument communication is started, and also:
```
assert os.path.exists(target_directory)
```